### PR TITLE
feat: implement calculator elements for alteration handling (HL-1155)

### PR DESCRIPTION
**Pull request by EmiliaMakelaVincit**

### CODE REVIEW

File: frontend/benefit/handler/src/components/applicationReview/calculationTable/useCalculationTable.ts [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+    (acc: number, cur: BenefitRow) => acc + parseFloat(cur.amountNumber),
+    0
+  );
+
+  if (tableRows.length > 0)
+    tableRows.push({
+      ...createBenefitRow(),
+      id: 'table-footer',
+      dates: `${tableRows.at(0).startDate} - ${tableRows.at(-1).endDate}`,
+      // Summing up multiple duration periods may cause float number inaccuracies to accumulate;
```
Comment (EmiliaMakelaVincit) [inline frontend/benefit/handler/src/components/applicationReview/calculationTable/useCalculationTable.ts:88]: Note: while most of this file is moved as-is from CalculationReview, this formatting part here is new in this PR.

**Comment by terovirtanen** [comment]
**API branch is deployed to platta: https://helsinkilisa-pr3052.api.dev.hel.ninja/healthz** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**APPLICANT branch is deployed to platta: https://helsinkilisa-ui-pr3052.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://helsinkilisa-ui-pr3052.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**HANDLER branch is deployed to platta: https://helsinkilisa-ui-handler-pr3052.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://helsinkilisa-ui-pr3052.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**API branch is deployed to platta: https://helsinkilisa-pr3052.api.dev.hel.ninja/healthz** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**HANDLER branch is deployed to platta: https://helsinkilisa-ui-handler-pr3052.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://helsinkilisa-ui-pr3052.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
## TestCafe handler result is __failed__ for https://helsinkilisa-ui-handler-pr3052.dev.hel.ninja :crying_cat_face::anger::boom::boom:

**Review by sirtawast — APPROVED** [review]
A lot of this happening, so that can't really follow in one commit. Refactoring towards reusability is nice to see :)

**Comment by terovirtanen** [comment]
**API branch is deployed to platta: https://helsinkilisa-pr3052.api.dev.hel.ninja/healthz** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**HANDLER branch is deployed to platta: https://helsinkilisa-ui-handler-pr3052.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**APPLICANT branch is deployed to platta: https://helsinkilisa-ui-pr3052.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://helsinkilisa-ui-pr3052.dev.hel.ninja :laughing::tada::tada::tada:

**Pull request merged**